src/amo/components/CollectionManager/index.js

```diff
@@ -183,6 +185,21 @@ export class CollectionManagerBase extends React.Component<Props, State> {
     }));
   };
 
+
+  handleSlug = (targetElement: Object) => {
+    const { name = '', value = '' } = targetElement;
+    const { creating } = this.props;
+    // only for add collection
+    if (!creating) return true;
+
+    if (name === 'name' && !this.state.userEnterSlug) {
+      this.setState({ slug: value.replace(/[^A-Za-z0-9]/g, '-') });
+    } else if (name === 'slug' && value.trim() !== '') {
+      this.setState({ userEnterSlug: true });
+    }
+    return true;
+  }
```

Comment:
Maybe this would be better in the `onTextInput` method so that we could only call `setState` one time.